### PR TITLE
#25 fixing routing error for signout link

### PR DESCRIPTION
**Pull request by Sontsevolosa**
* Adding 'capybara', 'capybara-screenshot', 'poltergeist', 'phantomjs' gems to the Gemfile (and adding some of them to the test helper file)
* Adding an integration test for checking 'Sign out' link

### CODE REVIEW

**Comment by Sontsevolosa** [comment]
@mluts please, review my code

File: test/integration/signout_button_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class SignoutButtonTest < ActionDispatch::IntegrationTest
+  test 'sign out button' do
+    visit('/')
+    sign_in users(:default)
+    click_button('Log in')
+    visit('/')
+    screenshot_and_save_page
+    click_on('Sign out')
+    assert page.has_content?('Log in'), 'Your page does not have content'
```
Comment (mluts) [inline test/integration/signout_button_test.rb]: Could you use path helper instead of `'/'` ?

Comment (mluts) [inline test/integration/signout_button_test.rb]: Also, there is `assert_text` matcher for this https://github.com/jnicklas/capybara/blob/2.7.1/lib/capybara/node/matchers.rb#L521

Comment (mluts) [inline test/integration/signout_button_test.rb]: @Sontsevolosa also, please add some newlines to test to break it down to logical parts

**Comment by mluts** [comment]
@Sontsevolosa please consider my comments

File: test/integration/signout_button_test.rb (hunk 1 above)
Comment (Sontsevolosa) [inline test/integration/signout_button_test.rb]: > Also, there is assert_text matcher for this https://github.com/jnicklas/capybara/blob/2.7.1/lib/capybara/node/matchers.rb#L521

@mluts Should I add 'assert_text' to the test? Did you mean this?

Comment (Sontsevolosa) [inline test/integration/signout_button_test.rb]: > also, please add some newlines to test to break it down to logical parts

@mluts I don't understand where I should add lines... In my test one line = one action. What did you mean?

Comment (mluts) [inline test/integration/signout_button_test.rb]: > Should I add 'assert_text' to the test? Did you mean this?
yes

>  I don't understand where I should add lines... In my test one line = one action. What did you mean?

For example:

```ruby
    visit('/')
    sign_in users(:default)
    click_button('Log in')

    visit('/')
    screenshot_and_save_page
    click_on('Sign out')

    assert page.has_content?('Log in'), 'Your page does not have content'
```

Comment (mluts) [inline test/integration/signout_button_test.rb]: @Sontsevolosa also, you don't need `screenshot_and_save_page` each time this test run, you will have 100500 screenshots in future

**Comment by Sontsevolosa** [comment]
@mluts please, review my changes

**Comment by Sontsevolosa** [comment]
@mluts please, review my new code

**Comment by Sontsevolosa** [comment]
@mluts please, review my changes

File: test/integration/signout_button_test.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+class SignoutButtonTest < ActionDispatch::IntegrationTest
+  test 'sign out button' do
+    visit root_path
+    sign_in users(:default)
+    click_button('Log in')
+    assert_text('Hello everyone!')
+
+    visit root_path
+    #screenshot_and_save_page
```
Comment (mluts) [inline test/integration/signout_button_test.rb]: @Sontsevolosa please remove the comment

File: test/integration/signout_button_test.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    click_on('Sign out')
+    assert page.has_content?('Log in'), 'Your page does not have content'
```
Comment (mluts) [inline test/integration/signout_button_test.rb]: @Sontsevolosa use assert_text

**Comment by Sontsevolosa** [comment]
@mluts please, review my changes

**Comment by mluts** [comment]
@Sontsevolosa PR is good, you can merge it

**Pull request merged**